### PR TITLE
check inc_dirs returned by nc-config for hdf5

### PR DESCRIPTION
**Pull request by jswhit**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -346,7 +357,7 @@ def _populate_hdf5_info(dirstosearch, inc_dirs, libs, lib_dirs):
         else:
             if HDF5_incdir is None:
                 HDF5_incdir = os.path.join(HDF5_dir, 'include')
-            hdf5_version = check_hdf5version(HDF5_incdir)
+            hdf5_version = get_hdf5version(HDF5_incdir)
```
Comment (sroet) [inline setup.py:360]: this should have an underscore
```suggestion
            hdf5_version = get_hdf5_version(HDF5_incdir)
```

File: setup.py [hunk 2]
```diff
@@ -331,8 +343,7 @@ def _populate_hdf5_info(dirstosearch, inc_dirs, libs, lib_dirs):
             sys.stdout.write("""
     HDF5_DIR environment variable not set, checking some standard locations ..\n""")
             for direc in dirstosearch:
-                sys.stdout.write('checking %s ...\n' % direc)
-                hdf5_version = check_hdf5version(os.path.join(direc, 'include'))
+                hdf5_version = get_hdf5version(os.path.join(direc, 'include'))
```
Comment (sroet) [inline setup.py:346]: This should have an underscore
```suggestion
                hdf5_version = get_hdf5_version(os.path.join(direc, 'include'))
```

**Comment by sroet** [comment]
This solves the issue I was trying to solve with #1029 

After fixing the above name errors it still fails for the non-default examples as it still overwrites env variables

However, the other examples still fail, it still overwrites env variables
<details><summary>Example: Linking with HDF5_DIR to 1.8.16 but netcdf4 uses 1.10.6 anyway. (click me)</summary>
<p>

```
# start in a the clone of this repo (master branch)
# create a conda env with hdf 1.8.16, this can be skipped with a preinstalled version in one of the standard directories
conda create -n hdf hdf5==1.8.16 -y
# Now create an environment for testing and activate
conda create -n test python=3 -y
conda activate test

# Install cython and netcdf4 dependencies
conda install -y cython
conda install -y --only-deps netcdf4

# Show that it now installed hdf=1.10.6
conda list hdf5

# Now we are going to install netcdf4, but explicitly point HDF5_DIR to the 1.8.16 conda environment (this can be done without HDF5_DIR if hdf5 is installed in one of the standard dirs)
HDF5_DIR=$CONDA_PREFIX/../hdf pip install -e .

# Let's see what is picked up in the test
cd test
python run_all.py
```

expected output:
```
netcdf4-python version: 1.5.4
HDF5 lib version:       1.8.16
netcdf lib version:     4.7.4
numpy version           1.19.0
cython version          0.29.21
```

Actual output:
```
netcdf4-python version: 1.5.4
HDF5 lib version:       1.10.6
netcdf lib version:     4.7.4
numpy version           1.19.0
cython version          0.29.21
```

</p>
</details>

It still doesn't link properly to stuff outside (after fixing the name errors I gave on the review of #1032), the easy way is probably to make sure we inspect `PATH` and raise a warning that a user should prepend their HDF_DIR to that path

<details><summary> Even more broken behavior if hdf5 is not present in the env during build-time and run-time (click me)</summary>
<p>

Starting from the end of the previous example (conda environment `test` is active, cwd is netcdf4-python/test) 
```
# Go back to main directory
cd ../

# break environment by force removing hdf5
conda remove hdf5 --force -y

# install netcdf4 again, HDF5_DIR is not needed if hdf5 is already installed in one of the standard dirs
HDF5_DIR=$CONDA_PREFIX/../hdf pip install -e .

# Let's see if we can run the test suite (with the HDF5_DIR variable)
cd test
HDF5_DIR=$CONDA_PREFIX/../hdf python run_all.py
```
This raises an import error

```
Traceback (most recent call last):
  File "run_all.py", line 2, in <module>
    from netCDF4 import getlibversion,__hdf5libversion__,__netcdf4libversion__,__version__
  File "/home/sroet/github_files/netcdf4-python/netCDF4/__init__.py", line 3, in <module>
    from ._netCDF4 import *
ImportError: libhdf5_hl.so.100: cannot open shared object file: No such file or directory
```

</p>
</details>

It does search PATH after reinstalling it, leading to HDF5 dumping it's core
<details><summary> HDF5 dumps core after we install another hdf in PATH (click me)</summary>
<p>

Continue the previous example (broken by force removing hdf5, cwd=netcdf4-python/test) by installing hdf5 but not rebuilding netcdf4.

```
conda install -y hdf5
python run_all.py
```

output:
```
netcdf4-python version: 1.5.4
HDF5 lib version:       1.8.16
netcdf lib version:     4.7.4
numpy version           1.19.1
cython version          0.29.21
.Warning! ***HDF5 library version mismatched error***
The HDF5 header files used to compile this application do not match
the version used by the HDF5 library to which this application is linked.
Data corruption or segmentation faults may occur if the application continues.
This can happen when an application was compiled by one version of HDF5 but
linked with a different version of static or shared HDF5 library.
You should recompile the application or check your shared library related
settings such as 'LD_LIBRARY_PATH'.
You can, at your own risk, disable this warning by setting the environment
variable 'HDF5_DISABLE_VERSION_CHECK' to a value of '1'.
Setting it to 2 or higher will suppress the warning messages totally.
Headers are 1.10.6, library is 1.8.16
	    SUMMARY OF THE HDF5 CONFIGURATION
	    =================================

General Information:
-------------------
		   HDF5 Version: 1.8.16
		  Configured on: Sun Aug  7 21:39:57 UTC 2016
		  Configured by: root@08a75980133e
		 Configure mode: production
		    Host system: x86_64-unknown-linux-gnu
	      Uname information: Linux 08a75980133e 3.13.0-91-generic #138-Ubuntu SMP Fri Jun 24 17:00:34 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
		       Byte sex: little-endian
		      Libraries: static, shared
	     Installation point: /home/sander/miniconda3/envs/hdf

Compiling Options:
------------------
               Compilation Mode: production
                     C Compiler: /opt/rh/devtoolset-2/root/usr/bin/gcc ( gcc (GCC) 4.8.2 20140120 )
                         CFLAGS: 
                      H5_CFLAGS: -std=c99 -pedantic -Wall -Wextra -Wundef -Wshadow -Wpointer-arith -Wbad-function-cast -Wcast-qual -Wcast-align -Wwrite-strings -Wconversion -Waggregate-return -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations -Wredundant-decls -Wnested-externs -Winline -Wfloat-equal -Wmissing-format-attribute -Wmissing-noreturn -Wpacked -Wdisabled-optimization -Wformat=2 -Wunreachable-code -Wendif-labels -Wdeclaration-after-statement -Wold-style-definition -Winvalid-pch -Wvariadic-macros -Winit-self -Wmissing-include-dirs -Wswitch-default -Wswitch-enum -Wunused-macros -Wunsafe-loop-optimizations -Wc++-compat -Wstrict-overflow -Wlogical-op -Wlarger-than=2048 -Wvla -Wsync-nand -Wframe-larger-than=16384 -Wpacked-bitfield-compat -Wstrict-overflow=5 -Wjump-misses-init -Wdouble-promotion -Wsuggest-attribute=const -Wtrampolines -Wstack-usage=8192 -Wvector-operation-performance -Wsuggest-attribute=pure -Wsuggest-attribute=noreturn -Wsuggest-attribute=format -O3
                      AM_CFLAGS: 
                       CPPFLAGS: 
                    H5_CPPFLAGS: -D_GNU_SOURCE -D_POSIX_C_SOURCE=200112L   -DNDEBUG -UH5_DEBUG_API
                    AM_CPPFLAGS: -D_LARGEFILE64_SOURCE -D_LARGEFILE_SOURCE  -I/home/sander/miniconda3/envs/hdf/include
               Shared C Library: yes
               Static C Library: yes
  Statically Linked Executables: no
                        LDFLAGS: 
                     H5_LDFLAGS: 
                     AM_LDFLAGS:  -L/home/sander/miniconda3/envs/hdf/lib
 	 	Extra libraries: -lrt -lz -ldl -lm 
 		       Archiver: ar
 		 	 Ranlib: ranlib
 	      Debugged Packages: 
		    API Tracing: no

Languages:
----------
                        Fortran: no

                            C++: yes
                   C++ Compiler: /opt/rh/devtoolset-2/root/usr/bin/g++ ( g++ (GCC) 4.8.2 20140120 )
                      C++ Flags: 
                   H5 C++ Flags:  
                   AM C++ Flags: 
             Shared C++ Library: yes
             Static C++ Library: yes

Features:
---------
                  Parallel HDF5: no
             High Level library: yes
                   Threadsafety: no
            Default API Mapping: v18
 With Deprecated Public Symbols: yes
         I/O filters (external): deflate(zlib)
                            MPE: no
                     Direct VFD: no
                        dmalloc: no
Clear file buffers before write: yes
           Using memory checker: no
         Function Stack Tracing: no
      Strict File Format Checks: no
   Optimization Instrumentation: no
Bye...
Aborted (core dumped)
```

</p>
</details>

What I would propose is that whenever HDF5_DIR is set we check that it is:
1. In `PATH`
2. The first in `PATH` to contain HDF5_DIR headers

and raise an informative error if this is not the case (or maybe static link instead?)

As that is apparently PATH is where  python (or HDF5?) searches for the headers during runtime

What would you think of that? (We can separate out the discussion for this to another issue/PR if you want)

**Comment by jswhit** [comment]
Not sure I am yet understanding the root cause of the problems you are seeing.  However, I've modified setup.py to always try to use nc-config first.  It will look in $NETCDF4_DIR/bin first, then use $PATH to find nc-config.  Since nc-config should provide the paths to the HDF5 library that netcdf was build with I believe this should solve the problem.  Can you give the PR a try again?

**Comment by sroet** [comment]
Alright with the current state of this branch:
### Default developer install

<details> <summary> Default developer install [works]</summary>
<p>

```
conda create -n test python=3
conda activate test
conda install cython
conda intstall --only-deps netcdf4
pip install -e .
cd test
python run_all.py
```
</p>
</details>

### Using env variable HDF5_DIR that is not in PATH

This might be intentional (overwriting HDF5_DIR), but should be documented somewhere.
<details> <summary> Linking to a different HDF5_DIR than the one in PATH [unexpected, works] </summary>
<p>

```
conda create -n hdf5 hdf5==1.8.16 
HDF5_DIR=$CONDA_PREFIX/../hdf5 pip install -e .
cd test
python run_all.py
```
this links to 1.10.6 instead of 1.8.16

```
not running tst_unicode.py ...

netcdf4-python version: 1.5.4
HDF5 lib version:       1.10.6
netcdf lib version:     4.7.4
numpy version           1.19.1
cython version          0.29.21
..................foo_bar
...........................................................................
----------------------------------------------------------------------
Ran 93 tests in 12.968s

OK
```

</p>
</details>

The following should probably fail on build-time instead of runtime (check that whatever we end up with after setup is still reachable)
Or should force static linking
<details> <summary> Linking to HDF5_DIR, without any HDF5 in PATH [broken]</summary>
<p>

Start from the previous setup (1 active env, named test with all dependencies and a env, named hdf5, that contains hdf5=1.8.16).
Break the env, install with HDF5_DIR, run test
```
conda remove hdf5 --force -y
HDF5_DIR=$CONDA_PREFIX/../hdf5 pip install -e .
cd test
python run_all.py
```
this fails on runtime
```
Traceback (most recent call last):
  File "run_all.py", line 2, in <module>
    from netCDF4 import getlibversion,__hdf5libversion__,__netcdf4libversion__,__version__
  File "/home/sander/github_files/netcdf4-python/netCDF4/__init__.py", line 3, in <module>
    from ._netCDF4 import *
ImportError: libhdf5_hl.so.100: cannot open shared object file: No such file or directory
```

</p>
</details>

Because the previous does not fail on build-time the following is possible
<details><summary> Add a different HDF5 to PATH after build [broken, core dump]</summary>
<p>

Start from the previous (netcdf4 build agains hdf5=1.8.16, but not in PATH).
Now install hdf5 again (adding it to PATH), and run the test again

```
conda install hdf5
python run_all.py
```
this returns
```
netcdf4-python version: 1.5.4
HDF5 lib version:       1.8.16
netcdf lib version:     4.7.4
numpy version           1.19.1
cython version          0.29.21
.Warning! ***HDF5 library version mismatched error***
The HDF5 header files used to compile this application do not match
the version used by the HDF5 library to which this application is linked.
Data corruption or segmentation faults may occur if the application continues.
This can happen when an application was compiled by one version of HDF5 but
linked with a different version of static or shared HDF5 library.
You should recompile the application or check your shared library related
settings such as 'LD_LIBRARY_PATH'.
You can, at your own risk, disable this warning by setting the environment
variable 'HDF5_DISABLE_VERSION_CHECK' to a value of '1'.
Setting it to 2 or higher will suppress the warning messages totally.
Headers are 1.10.6, library is 1.8.16
	    SUMMARY OF THE HDF5 CONFIGURATION
	    =================================

General Information:
-------------------
		   HDF5 Version: 1.8.16
		  Configured on: Sun Aug  7 21:39:57 UTC 2016
		  Configured by: root@08a75980133e
		 Configure mode: production
		    Host system: x86_64-unknown-linux-gnu
	      Uname information: Linux 08a75980133e 3.13.0-91-generic #138-Ubuntu SMP Fri Jun 24 17:00:34 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
		       Byte sex: little-endian
		      Libraries: static, shared
	     Installation point: /home/sander/miniconda3/envs/hdf5

Compiling Options:
------------------
               Compilation Mode: production
                     C Compiler: /opt/rh/devtoolset-2/root/usr/bin/gcc ( gcc (GCC) 4.8.2 20140120 )
                         CFLAGS: 
                      H5_CFLAGS: -std=c99 -pedantic -Wall -Wextra -Wundef -Wshadow -Wpointer-arith -Wbad-function-cast -Wcast-qual -Wcast-align -Wwrite-strings -Wconversion -Waggregate-return -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations -Wredundant-decls -Wnested-externs -Winline -Wfloat-equal -Wmissing-format-attribute -Wmissing-noreturn -Wpacked -Wdisabled-optimization -Wformat=2 -Wunreachable-code -Wendif-labels -Wdeclaration-after-statement -Wold-style-definition -Winvalid-pch -Wvariadic-macros -Winit-self -Wmissing-include-dirs -Wswitch-default -Wswitch-enum -Wunused-macros -Wunsafe-loop-optimizations -Wc++-compat -Wstrict-overflow -Wlogical-op -Wlarger-than=2048 -Wvla -Wsync-nand -Wframe-larger-than=16384 -Wpacked-bitfield-compat -Wstrict-overflow=5 -Wjump-misses-init -Wdouble-promotion -Wsuggest-attribute=const -Wtrampolines -Wstack-usage=8192 -Wvector-operation-performance -Wsuggest-attribute=pure -Wsuggest-attribute=noreturn -Wsuggest-attribute=format -O3
                      AM_CFLAGS: 
                       CPPFLAGS: 
                    H5_CPPFLAGS: -D_GNU_SOURCE -D_POSIX_C_SOURCE=200112L   -DNDEBUG -UH5_DEBUG_API
                    AM_CPPFLAGS: -D_LARGEFILE64_SOURCE -D_LARGEFILE_SOURCE  -I/home/sander/miniconda3/envs/hdf5/include
               Shared C Library: yes
               Static C Library: yes
  Statically Linked Executables: no
                        LDFLAGS: 
                     H5_LDFLAGS: 
                     AM_LDFLAGS:  -L/home/sander/miniconda3/envs/hdf5/lib
 	 	Extra libraries: -lrt -lz -ldl -lm 
 		       Archiver: ar
 		 	 Ranlib: ranlib
 	      Debugged Packages: 
		    API Tracing: no

Languages:
----------
                        Fortran: no

                            C++: yes
                   C++ Compiler: /opt/rh/devtoolset-2/root/usr/bin/g++ ( g++ (GCC) 4.8.2 20140120 )
                      C++ Flags: 
                   H5 C++ Flags:  
                   AM C++ Flags: 
             Shared C++ Library: yes
             Static C++ Library: yes

Features:
---------
                  Parallel HDF5: no
             High Level library: yes
                   Threadsafety: no
            Default API Mapping: v18
 With Deprecated Public Symbols: yes
         I/O filters (external): deflate(zlib)
                            MPE: no
                     Direct VFD: no
                        dmalloc: no
Clear file buffers before write: yes
           Using memory checker: no
         Function Stack Tracing: no
      Strict File Format Checks: no
   Optimization Instrumentation: no
Bye...
Aborted (core dumped)
```


</p>
</details>

### Using the searched dir `/sw` with hdf5=1.8.16

<details><summary> Some setup to prep /sw </summary>
<p>

Copy everything from the hdf5 env to `/sw`

```
cp -r $CONDA_PREFIX/../hdf5/* /sw/
```

</p>
</details>

<details><summary> Install with both HDF5 in path and /sw [works] </summary>
<p>

```
pip install -e .
cd tests
pythun run_all.py
```

</p>
</details>


The following should probably behave the same as with a given HDF5_DIR (fail on build-time or static linking)

<details><summary>  Install without HDF5 in PATH but only in /sw [broken] </summary>
<p>

Break the path again, install, run tests
 
``` 
conda remove hdf5 --force -y
pip install -e .
cd test
python run_all.py

```
no headers are found
```
Traceback (most recent call last):
  File "run_all.py", line 2, in <module>
    from netCDF4 import getlibversion,__hdf5libversion__,__netcdf4libversion__,__version__
  File "/home/sander/github_files/netcdf4-python/netCDF4/__init__.py", line 3, in <module>
    from ._netCDF4 import *
ImportError: libhdf5_hl.so.100: cannot open shared object file: No such file or directory
```


</p>
</details>

This is again possible because the previous one does not break on build-time or force static linking

<details><summary> Add HDF5 back to PATH after building [broken, core dump]</summary>
<p>

We add hdf5 again, and run the tests
```
conda install hdf5
cd test
python run_all.py
```
this let hdf5 dump it's core
```
not running tst_unicode.py ...

netcdf4-python version: 1.5.4
HDF5 lib version:       1.8.16
netcdf lib version:     4.7.4
numpy version           1.19.1
cython version          0.29.21
.Warning! ***HDF5 library version mismatched error***
The HDF5 header files used to compile this application do not match
the version used by the HDF5 library to which this application is linked.
Data corruption or segmentation faults may occur if the application continues.
This can happen when an application was compiled by one version of HDF5 but
linked with a different version of static or shared HDF5 library.
You should recompile the application or check your shared library related
settings such as 'LD_LIBRARY_PATH'.
You can, at your own risk, disable this warning by setting the environment
variable 'HDF5_DISABLE_VERSION_CHECK' to a value of '1'.
Setting it to 2 or higher will suppress the warning messages totally.
Headers are 1.10.6, library is 1.8.16
	    SUMMARY OF THE HDF5 CONFIGURATION
	    =================================

General Information:
-------------------
		   HDF5 Version: 1.8.16
		  Configured on: Sun Aug  7 21:39:57 UTC 2016
		  Configured by: root@08a75980133e
		 Configure mode: production
		    Host system: x86_64-unknown-linux-gnu
	      Uname information: Linux 08a75980133e 3.13.0-91-generic #138-Ubuntu SMP Fri Jun 24 17:00:34 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
		       Byte sex: little-endian
		      Libraries: static, shared
	     Installation point: /home/sander/miniconda3/envs/hdf5

Compiling Options:
------------------
               Compilation Mode: production
                     C Compiler: /opt/rh/devtoolset-2/root/usr/bin/gcc ( gcc (GCC) 4.8.2 20140120 )
                         CFLAGS: 
                      H5_CFLAGS: -std=c99 -pedantic -Wall -Wextra -Wundef -Wshadow -Wpointer-arith -Wbad-function-cast -Wcast-qual -Wcast-align -Wwrite-strings -Wconversion -Waggregate-return -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations -Wredundant-decls -Wnested-externs -Winline -Wfloat-equal -Wmissing-format-attribute -Wmissing-noreturn -Wpacked -Wdisabled-optimization -Wformat=2 -Wunreachable-code -Wendif-labels -Wdeclaration-after-statement -Wold-style-definition -Winvalid-pch -Wvariadic-macros -Winit-self -Wmissing-include-dirs -Wswitch-default -Wswitch-enum -Wunused-macros -Wunsafe-loop-optimizations -Wc++-compat -Wstrict-overflow -Wlogical-op -Wlarger-than=2048 -Wvla -Wsync-nand -Wframe-larger-than=16384 -Wpacked-bitfield-compat -Wstrict-overflow=5 -Wjump-misses-init -Wdouble-promotion -Wsuggest-attribute=const -Wtrampolines -Wstack-usage=8192 -Wvector-operation-performance -Wsuggest-attribute=pure -Wsuggest-attribute=noreturn -Wsuggest-attribute=format -O3
                      AM_CFLAGS: 
                       CPPFLAGS: 
                    H5_CPPFLAGS: -D_GNU_SOURCE -D_POSIX_C_SOURCE=200112L   -DNDEBUG -UH5_DEBUG_API
                    AM_CPPFLAGS: -D_LARGEFILE64_SOURCE -D_LARGEFILE_SOURCE  -I/home/sander/miniconda3/envs/hdf5/include
               Shared C Library: yes
               Static C Library: yes
  Statically Linked Executables: no
                        LDFLAGS: 
                     H5_LDFLAGS: 
                     AM_LDFLAGS:  -L/home/sander/miniconda3/envs/hdf5/lib
 	 	Extra libraries: -lrt -lz -ldl -lm 
 		       Archiver: ar
 		 	 Ranlib: ranlib
 	      Debugged Packages: 
		    API Tracing: no

Languages:
----------
                        Fortran: no

                            C++: yes
                   C++ Compiler: /opt/rh/devtoolset-2/root/usr/bin/g++ ( g++ (GCC) 4.8.2 20140120 )
                      C++ Flags: 
                   H5 C++ Flags:  
                   AM C++ Flags: 
             Shared C++ Library: yes
             Static C++ Library: yes

Features:
---------
                  Parallel HDF5: no
             High Level library: yes
                   Threadsafety: no
            Default API Mapping: v18
 With Deprecated Public Symbols: yes
         I/O filters (external): deflate(zlib)
                            MPE: no
                     Direct VFD: no
                        dmalloc: no
Clear file buffers before write: yes
           Using memory checker: no
         Function Stack Tracing: no
      Strict File Format Checks: no
   Optimization Instrumentation: no
Bye...
Aborted (core dumped)
```


</p>
</details>

**Comment by jswhit** [comment]
Not sure what you mean by 'HDF5 in PATH'?  The env var $PATH is not used by setup.py anywhere, except to find the nc-config utility.

**Comment by sroet** [comment]
> Not sure what you mean by 'HDF5 in PATH'? The env var $PATH is not used by setup.py anywhere, except to find the nc-config utility.

The issue that I have is that `setup.py` can be used to create an install with libraries that can't be found again during runtime.

It seems like this might be more in-depth than a broken setup.py library link, as the file it is looking for does not even exist for `hdf5=1.8.16` (`libhdf5_hl.so.100`)  

<details> <summary> ls of the included lib dir [click me]</summary>
<p>

```
-rwxr-xr-x 1 root root 2,4M aug  3 14:36 libubsan.so.1.0.0
-rwxr-xr-x 1 root root 945K aug  3 14:36 libquadmath.so.0.0.0
-rwxr-xr-x 1 root root 940K aug  3 14:36 libitm.so.1.0.0
-rw-r--r-- 1 root root  132 aug  3 14:36 libgcc_s.so
-rwxr-xr-x 1 root root 183K aug  3 14:36 libatomic.so.1.2.0
-rwxr-xr-x 1 root root 6,8M aug  3 14:36 libtsan.so.0.0.0
-rwxr-xr-x 1 root root 2,6M aug  3 14:36 liblsan.so.0.0.0
-rw-r--r-- 1 root root 3,3K aug  3 14:36 libhdf5.settings
-rwxr-xr-x 1 root root 1002 aug  3 14:36 libhdf5.la
-rwxr-xr-x 1 root root 1,1M aug  3 14:36 libgomp.so.1.0.0
-rw-r--r-- 1 root root 808K aug  3 14:36 libgcc_s.so.1
-rwxr-xr-x 1 root root 9,8M aug  3 14:36 libasan.so.5.0.0
drwxr-xr-x 2 root root 4,0K aug  3 14:36 pkgconfig
lrwxrwxrwx 1 root root   14 aug  3 14:36 libz.so.1 -> libz.so.1.2.11
lrwxrwxrwx 1 root root   14 aug  3 14:36 libz.so -> libz.so.1.2.11
lrwxrwxrwx 1 root root   17 aug  3 14:36 libubsan.so.1 -> libubsan.so.1.0.0
lrwxrwxrwx 1 root root   17 aug  3 14:36 libubsan.so -> libubsan.so.1.0.0
lrwxrwxrwx 1 root root   16 aug  3 14:36 libtsan.so.0 -> libtsan.so.0.0.0
lrwxrwxrwx 1 root root   16 aug  3 14:36 libtsan.so -> libtsan.so.0.0.0
lrwxrwxrwx 1 root root   20 aug  3 14:36 libquadmath.so.0 -> libquadmath.so.0.0.0
lrwxrwxrwx 1 root root   20 aug  3 14:36 libquadmath.so -> libquadmath.so.0.0.0
lrwxrwxrwx 1 root root   16 aug  3 14:36 liblsan.so.0 -> liblsan.so.0.0.0
lrwxrwxrwx 1 root root   16 aug  3 14:36 liblsan.so -> liblsan.so.0.0.0
lrwxrwxrwx 1 root root   15 aug  3 14:36 libitm.so.1 -> libitm.so.1.0.0
lrwxrwxrwx 1 root root   15 aug  3 14:36 libitm.so -> libitm.so.1.0.0
-rwxr-xr-x 1 root root 3,2M aug  3 14:36 libhdf5.so.10.1.0
lrwxrwxrwx 1 root root   17 aug  3 14:36 libhdf5.so.10 -> libhdf5.so.10.1.0
lrwxrwxrwx 1 root root   17 aug  3 14:36 libhdf5.so -> libhdf5.so.10.1.0
-rwxr-xr-x 1 root root 142K aug  3 14:36 libhdf5_hl.so.10.0.2
lrwxrwxrwx 1 root root   20 aug  3 14:36 libhdf5_hl.so.10 -> libhdf5_hl.so.10.0.2
lrwxrwxrwx 1 root root   20 aug  3 14:36 libhdf5_hl.so -> libhdf5_hl.so.10.0.2
-rwxr-xr-x 1 root root 1,1K aug  3 14:36 libhdf5_hl.la
-rwxr-xr-x 1 root root  14K aug  3 14:36 libhdf5_hl_cpp.so.11.0.0
lrwxrwxrwx 1 root root   24 aug  3 14:36 libhdf5_hl_cpp.so.11 -> libhdf5_hl_cpp.so.11.0.0
lrwxrwxrwx 1 root root   24 aug  3 14:36 libhdf5_hl_cpp.so -> libhdf5_hl_cpp.so.11.0.0
-rwxr-xr-x 1 root root 1,2K aug  3 14:36 libhdf5_hl_cpp.la
-rw-r--r-- 1 root root 9,1K aug  3 14:36 libhdf5_hl_cpp.a
-rw-r--r-- 1 root root 212K aug  3 14:36 libhdf5_hl.a
-rwxr-xr-x 1 root root 512K aug  3 14:36 libhdf5_cpp.so.11.0.0
lrwxrwxrwx 1 root root   21 aug  3 14:36 libhdf5_cpp.so.11 -> libhdf5_cpp.so.11.0.0
lrwxrwxrwx 1 root root   21 aug  3 14:36 libhdf5_cpp.so -> libhdf5_cpp.so.11.0.0
-rwxr-xr-x 1 root root 1,1K aug  3 14:36 libhdf5_cpp.la
-rw-r--r-- 1 root root 844K aug  3 14:36 libhdf5_cpp.a
lrwxrwxrwx 1 root root   16 aug  3 14:36 libgomp.so.1 -> libgomp.so.1.0.0
lrwxrwxrwx 1 root root   16 aug  3 14:36 libgomp.so -> libgomp.so.1.0.0
lrwxrwxrwx 1 root root   18 aug  3 14:36 libatomic.so.1 -> libatomic.so.1.2.0
lrwxrwxrwx 1 root root   18 aug  3 14:36 libatomic.so -> libatomic.so.1.2.0
lrwxrwxrwx 1 root root   16 aug  3 14:36 libasan.so.5 -> libasan.so.5.0.0
lrwxrwxrwx 1 root root   16 aug  3 14:36 libasan.so -> libasan.so.5.0.0
-rwxr-xr-x 1 root root 107K aug  3 14:36 libz.so.1.2.11
-rw-r--r-- 1 root root 147K aug  3 14:36 libz.a
-rw-r--r-- 1 root root 5,8M aug  3 14:36 libhdf5.a
```

</p>
</details>

**Comment by jswhit** [comment]
The hdf5 shared library version that netcdf-c was linked to is encoded in the netcdf c shared library (you can see this by looking at the output of ldd).  That fact that this library version doesn't exist on your system or can't be found at runtime is not the fault of the netcdf4-python build.

**Comment by sroet** [comment]
> The hdf5 shared library version that netcdf-c was linked to is encoded in the netcdf c shared library (you can see this by looking at the output of ldd). That fact that this library version doesn't exist on your system or can't be found at runtime is not the fault of the netcdf4-python build.

Ah, that would explain it. Thanks for clearing that up, no further issues from me then.

**Pull request merged**